grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1737 @@
+# Copyright (c) 2019-2023, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+
+import warnings
+import numpy as np
+import unittest
+import os
+import tempfile
+from grid2op.Observation import BaseObservation
+from grid2op.tests.helper_path_test import *
+
+from grid2op import make
+from grid2op.Reward import _AlertTrustScore
+from grid2op.Parameters import Parameters
+from grid2op.Exceptions import Grid2OpException
+from grid2op.Runner import Runner
+from grid2op.Action import BaseAction, PlayableAction
+from grid2op.Agent import BaseAgent
+from grid2op.Episode import EpisodeData
+
+from _aux_opponent_for_test_alerts import (_get_steps_attack,
+                                           TestOpponent,
+                                           TestOpponentMultiLines)
+
+ATTACKED_LINE = "48_50_136"
+
+DEFAULT_PARAMS_TRUSTSCORE = dict(reward_min_no_blackout=-1.0,
+                                 reward_min_blackout=-50.0,
+                                 reward_max_no_blackout=0.0,
+                                 reward_max_blackout=0.0,
+                                 reward_end_episode_bonus=0.0,
+                                 min_score=-3.0)
+
+#a near copy of _normalisation_fun function from alertTrusScore. Use when, for to a given trustscore parametrization,
+# it is not easy to guess the score before hand, for a given scenario.
+# especially when reward_end_episode_bonus is non null for some non blackout cases
+def manual_score (cm_reward,cm_reward_min_ep,cm_reward_max_ep,max_score):
+
+    manual_standardized_score= np.round((cm_reward - cm_reward_min_ep) / (cm_reward_max_ep - cm_reward_min_ep + 1e-5), 4)
+    manual_score = DEFAULT_PARAMS_TRUSTSCORE["min_score"] + (
+                                max_score - DEFAULT_PARAMS_TRUSTSCORE[
+                            "min_score"]) * manual_standardized_score
+    return manual_score
+
+# Test alertTrustScore when no blackout and when blackout
+class TestAlertTrustScoreNoBlackout(unittest.TestCase):
```

Comment:
It's a a good start to have copy pasted all the tests in AlertReward and made them pass for this class. But it's not enough as i'm not sure it tests the new behaviour.

Especially, i don't thing you tried to find "corner cases" where the behaviour of the new class and the old one would differ. The previous tests were made for "corner cases" in the AlertReward. I'm not sure if there are the same "corner cases" for this class.

Reply:
(though it's good to make sure there are no regressions on the code anywhere)

Reply:
For example, the whole things is based on the  assumption that "whatever the case, the boundaries in `_normalisation_fun` can be achieved"

So it would be nice to include (or at least put together) tests that, for some "basic" situations (*eg* `blackout "due to" attack`, `blackout not due to attack`, `no blackout`) that you can design 3 agents : one achieving min score, one achieving max score and one performing "in the middle"